src/java/org/apache/cassandra/index/sai/disk/format/IndexFeatureSet.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.index.sai.disk.format;
+
+/**
+ * The {@code IndexFeatureSet} represents the set of features available is SAI that are not
+ * available in the {@code V1OnDiskFormat}. Version 1 features should not be added to here
+ * unless the feature is removed from a future version and Version 1 is still supported.
+ */
+public interface IndexFeatureSet
+{
+    /**
+     * The {@code Accumulator} is used to accumulate the {@code IndexFeatureSet} responses from
+     * multiple sources. This will include all the SSTables included in a query and all the indexes
+     * attached to those SSTables.
+     *
+     * The {@code Accumulator} creates an {@code IndexFeatureSet} this contains the features supported
+     * by the lowest version index involved in a query.
+     */
```

Comment:
What if a newer version of an index *removes* a feature? This documentation says that this would return the feature set of the index with the lowest version, so it might return a features set not fully supported by some newer indexes. 
If index v1 supports featres {A, B} and v2 supports features {A, C}, and we have both versions in the sstable set, the docs says it would return feature set {A, B}, not {A}. Why?

I guess there are two ways out:
- we never remove features
- we change the contract to return the intersection of features of all the indexes attached to the sstables

Reply:
Agreed, I think that comment is misleading. As I have said above they are simple `boolean` features with `false` taking precedence. 

If a feature is removed in a future version then is a mixed version query the value of the future version would take precedence because it would be `false`.

I'll try and revise the comment to make it clearer.